### ISSUE TITLE
[Issue] `ags run-js 'openMusicControls...'` just return `undefined`

### ISSUE THREAD

**Issue by clsty** [end-4/dots-hyprland#168]
- **I have read the instructions** and I have a problem with branch: 
---
## The issue
- The following keybind seems do nothing on my PC even when music is playing (and the bar of ags is showing that properly):
```conf
bind = Super, M, exec, ags run-js 'openMusicControls.value = (!Mpris.getPlayer() ? false : !openMusicControls.value);'
```
And when I run the command in terminal:
```bash
ags run-js 'openMusicControls.value = (!Mpris.getPlayer() ? false : !openMusicControls.value);'
```
the output is `undefined`.

**Comment by end-4** [comment]
- ags run-js always returns undefined
- does your whatever music app have mpris integration?

**Comment by clsty** [comment]
I'm using firefox. [Arch Wiki](https://wiki.archlinux.org/title/MPRIS) says it is one of supported MPRIS clients.

**Comment by end-4** [comment]
- Do you have plasma browser integration browser extension and package? Get them if not. Note that you might need to run the host process manually or relog
- You can install `playerctl` and run `playerctl -F metadata` to check if there's sufficient data. It should show a bunch of `plasma-browser-integration` properties
![image](https://github.com/end-4/dots-hyprland/assets/97237370/7cc4ce43-cc36-4106-83b3-56a22d090817)

**Comment by end-4** [comment]
![image](https://github.com/end-4/dots-hyprland/assets/97237370/06eab423-9288-406c-b3c2-42d9061b56e8)

**Comment by clsty** [comment]
> Do you have plasma browser integration browser extension and package?

Yes, I already have it both as firefox extension enabled and as Arch package `plasma-browser-integration` installed.

> run playerctl -F metadata to check if there's sufficient data. It should show a bunch of plasma-browser-integration properties

Here's output:
```bash
firefox mpris:trackid             '/org/mpris/MediaPlayer2/firefox'
firefox xesam:title               <The title of page playing the music, sorry I have to remove this info>
firefox xesam:album               
firefox xesam:artist
```

And for the screenshot you provided, I get it that I'm missing the `org.mpris.MediaPlayer2.plasma-browser...` entry in dbus, but I don't know why as I did already have both the extension and the package.

I'll try clementine (another supported mpris client) and a new profile for firefox later.

**Comment by clsty** [comment]
Oh yeah, for clementine the music panel (as well the keybinding `super+M`) works.

Now the problem falls on firefox.

**Comment by end-4** [comment]
can you try firefox + youtube?
